docs/testing/runtimes.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Runtimes in Office Add-ins
+description: Learn about the JavaScript runtimes that are used by Office Add-ins.
+ms.date: 08/10/2022
+ms.localizationpriority: medium
+---
+
+# Runtimes in Office Add-ins
+
+As an interpreted language, JavaScript must run in a JavaScript engine. As a single-threaded, synchronous language, JavaScript has no inherent capacity for asynchronous execution; but modern JavaScript engines can request asynchronous operations (including networking communication) from the host operating system and receive data from the OS in response. This kind of engine makes JavaScript *effectively* asynchronous. In this article, engines of this sort are called *runtimes*. [Node.js](https://nodejs.org) and modern browsers are examples of such runtimes. 
+
+## Types of runtimes
+
+There are two types of runtimes used by Office Add-ins:
+
+- **JavaScript-only runtime**: A JavaScript engine supplemented with support for [WebSockets](https://developer.mozilla.org/docs/Web/API/WebSockets_API), [CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/docs/Web/HTTP/CORS), and [local storage](https://developer.mozilla.org/docs/Web/API/Window/localStorage). 
```

Comment:
Should we say Full CORS to be specific since CORS could be Simple or Full, and the latter includes the former?

Reply:
AFAIK, we have only implemented Simple CORS. Full CORS has been pending for years. Alex can correct me.

Reply:
We should probably link to our own ReactNative docs, since we haven't implemented the complete Mozilla specs, AFAIK.

Reply:
> We should probably link to our own ReactNative docs, since we haven't implemented the complete Mozilla specs, AFAIK.

I think that would be confusing because we'd be linking to a framework that _does_ have a view. Also, we say later in the detailed section about the JS-only runtime that it is modified from the original RN.

Reply:
> AFAIK, we have only implemented Simple CORS. Full CORS has been pending for years. Alex can correct me.

We used to support Simple CORS. Now, we support Full CORS. I just sent you an email about this.